cmd/node/flags.go

```diff
@@ -405,14 +417,17 @@ func getFlags() []cli.Flag {
 		serializeSnapshots,
 		noKey,
 		p2pKeyPemFile,
+		dbDirectory,
+		logsDirectory,
 	}
 }
 
 func getFlagsConfig(ctx *cli.Context, log logger.Logger) *config.ContextFlagsConfig {
 	flagsConfig := &config.ContextFlagsConfig{}
 
-	workingDir := ctx.GlobalString(workingDirectory.Name)
-	flagsConfig.WorkingDir = getWorkingDir(workingDir, log)
+	flagsConfig.WorkingDir = getWorkingDir(ctx, workingDirectory, log)
+	flagsConfig.DbDir = getWorkingDir(ctx, dbDirectory, log)
+	flagsConfig.LogsDir = getWorkingDir(ctx, logsDirectory, log)
```

Comment:
what happens if? 
`--working-directory new-path`
`--db-path` not set
`--log-path` not set

it would save only the statistics on `new-path` but `db` and `logs` will stain near the node binary, while they should be inside `new-path` as well. so, only if the new flags are set they should change the behaviour in some way
This is not backward compatible

Reply:
right, thanks